tests/layer_and_mask_information_section.rs

```diff
@@ -66,12 +66,27 @@ fn two_groups_two_layers_inside() {
     let psd = Psd::from_bytes(psd).unwrap();
 
     // 2 group layer
-    assert_eq!(psd.layers().len(), 2);
+    assert_eq!(psd.groups().len(), 2);
 
     // Check first group
-    let group = psd.layer_by_name("group").unwrap();
-    group.layers().unwrap().layer_by_name("First Layer").unwrap();
+    psd.group_by_name("group").unwrap();
     // Check second group
-    let group = psd.layer_by_name("group2").unwrap();
-    group.layers().unwrap().layer_by_name("Second Layer").unwrap();
+    psd.group_by_name("group2").unwrap();
+}
+
+#[test]
+fn one_group_inside_another() {
+    let psd = include_bytes!("fixtures/green-1x1-one-group-inside-another.psd");
+    let psd = Psd::from_bytes(psd).unwrap();
+
+    assert_eq!(psd.layers().len(), 1);
+    // parent group + children group
+    assert_eq!(psd.groups().len(), 2);
+
+    // Check group
+    let group = psd.group_by_name("group outside").unwrap();
```

Comment:
Same here - can we verify that one group points to no layers and the other group points to one layer

Reply:
Also can we verify that the child group has its parent group set properly

Something like (just a rough sketch):

```
let parent = psd.group_by_name("group_inside").unwrap().parent_group().unwrap();
assert_eq!(psd.group_by_id(parent).unwrap().name, "group_outside")`
```

Reply:
Otherwise these tests aren't verifying that we're pulling the correct group data - just that we're pulling groups at all